mitodl/mitxpro
# User menu

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #439 

#### What's this PR do?
- Adds a `UserMenu` component in the `TopAppBar` for logged-in users.

#### How should this be manually tested?
- Browse the site when not logged in, the top of each page should look the same as before:

![Screen Shot 2019-06-14 at 3 30 09 PM](https://user-images.githubusercontent.com/187676/59533407-5d7bd000-8eb9-11e9-8228-f32408fb691e.png)


- Log in.  It should look like this:

![Screen Shot 2019-06-14 at 3 30 53 PM](https://user-images.githubusercontent.com/187676/59533432-6f5d7300-8eb9-11e9-8fd8-df0e6c531968.png)


- Click the small dropdown arrow near the default avatar icon.  You should see this dropdown of links and the links should all work:

### Desktop
![Screen Shot 2019-06-14 at 3 16 00 PM](https://user-images.githubusercontent.com/187676/59533450-7a180800-8eb9-11e9-9d60-caf59f54703c.png)

### Mobile
![Screen Shot 2019-06-14 at 3 22 41 PM](https://user-images.githubusercontent.com/187676/59533456-800de900-8eb9-11e9-98a1-6810e3aed2a5.png)


#### Any background context you want to provide?
- There is supposed to be a `Settings` link in the dropdown but there is no page for that yet.
~- I need to get the real icons for the dropdown items, for now the icons are from a screenshot of the invision mockup.~
